FlowCrypt/Controllers/Threads/MessageThread.swift

```diff
@@ -37,7 +37,7 @@ struct MessageThread: Equatable {
         guard let firstMessageLabels = messages.first?.labels else {
             return false
         }
-        // Thread is treaded as archived when first message label doesn't contain inbox & sent label
+        // Thread is treaded as archived when labels don't contain `inbox` and first message label doesn't contain sent label
```

Comment:
Looks like it fails in this situation:
1) receive message
2) archive it
3) go to 'All Mail' and send reply to this message
4) go to 'Sent' - this thread is visible (not archived, which is right), but `isArchived` is `true` (should be `false`)

Reply:
Well, `isArchived` doesn't seem to correct word but value should be `true` I think.
So that it can show `move to inbox` button for this thread and this thread moves to `inbox` when tapped.

Reply:
Maybe field name should be `shouldShowMoveToInboxButton`?

Reply:
let me know your thoughts

Reply:
I checked Gmail app - for this message it doesn't show `move to inbox` and `move to archive` buttons, just `delete`.
It works the same in our `master` branch, but in this branch it shows `move to inbox`, which is wrong.

Reply:
Ah, Gmail app shows `Move to inbox` action only when you open this message in `All Mail` folder, but not in `Sent` folder

Reply:
> Maybe field name should be `shouldShowMoveToInboxButton`?

This field name sounds good for me, but we need to update it's logic to work the same as in Gmail app (show `move to inbox` folder depending on current folder - `sent` or `all mail`)

Reply:
Hmm.. For above case in my gmail(browser version), it shows `archive` button in both `sent` and `all mail` folder (not showing `move to inbox` button)
But `archive` button doesn't have any effect in gmail app and seems like a bug.

Reply:
For above case, I think we should show `move to inbox` button for this thread as currently implemented(as thread is archived).

Reply:
Are you testing in gmail browser version or mobile app version?

Reply:
I check Gmail iOS app

Reply:
I see. Let me show `move to inbox` button only for `all mail` folder.

Reply:
> I think there's nothing wrong to show `move to inbox` button for this thread (even for sent folder)
> What do you suggest?

Okay, let's leave `move to inbox` button on `sent` screen, so we just need to rename property to `shouldShowMoveToInboxButton`

Reply:
okay